src/rebar_prv_install_deps.erl

```diff
@@ -510,7 +512,7 @@ parse_goal(Name, Constraint) ->
         {match, [Op, Vsn]} ->
             {Name, Vsn, binary_to_atom(Op, utf8)};
         nomatch ->
-            fail
+            throw(?PRV_ERROR({bad_constraint, Name, Constraint}))
```

Comment:
https://github.com/rebar/rebar3/pull/270/files#diff-420b69ef9b3a1a1b77933c8f1ee163c7L51 is the only one I broke I think.